transport_plugins/jlink/iotile_transport_jlink/jlink_background.py

```diff
@@ -161,8 +168,118 @@ def _trigger_rpc(self, device_info):
         else:
             raise HardwareError("Unknown RPC trigger method", method=method)
 
-    def _dump_all_ram(self, device_info, _control_info, _args, _progress_callback):
-        memory = self._read_memory(device_info.ram_start, device_info.ram_size)
+    def _continue(self):
+        self._jlink.step()
+        self._jlink._dll.JLINKARM_Go()
+        while not self._jlink.halted():
+            pass
+
+    def _inject_blob(self, rel_path_to_bin):
+        ffd_bin = pkg_resources.resource_string(__name__, rel_path_to_bin)
+        self._jlink.memory_write8(ffd_cfg.ffd_ram_addrs['inject_start'], ffd_bin)
+
+    def _update_reg(self, register, value):
+        if not self._jlink.halted():
+            self._jlink.halt()
+        self._jlink.register_write(register, value)
+
+    def _get_register_handle(self, register_string):
+        reg_list = self._jlink.register_list()
+        reg_str_paren = '(' + register_string + ')'
+
+        for reg in reg_list:
+            reg_name = self._jlink.register_name(reg)
+            if reg_name is register_string or reg_str_paren in reg_name:
+                return reg
+        return None
+
+    def _get_ffd_max_read_length(self):
+        max_read_length = self._jlink.memory_read16(ffd_cfg.ffd_ram_addrs['max_read_length'], 1)
+        return max_read_length[0]
+
+    def _write_ffd_dump_cmd(self, start, length):
```

Comment:
I know these are all private functions, and don't require docstrings, but I'd recommend doing another pass and making sure there aren't comments or docstrings you'll want to leave for your future self should you ever have to revisit :) 

Not a requirement to have docstrings on these private functions, though.

Reply:
Fixed. Removed comments, will be up in next push.